### ISSUE TITLE
[Bug] puppeteer-extra-plugin-recaptcha not working and access being blocked by Nike invisible reCAPTCHA

### ISSUE THREAD

**Issue by igorthurow** [berstend/puppeteer-extra#406]
**Describe the bug**
Hey guys! 
Always when i load the Nike sneaker product page from the puppeteer, the invisible reCAPTCHA blocks me very very fast and i do not have possibility to resolve i guess.

![image](https://user-images.githubusercontent.com/36999154/105612206-5ad1a400-5d99-11eb-9ac5-270592aa133c.png)

**Code Snippet**

```javascript
const puppeteer = require('puppeteer-extra')
const StealthPlugin = require('puppeteer-extra-plugin-stealth')
const RecaptchaPlugin = require('puppeteer-extra-plugin-recaptcha')
puppeteer.use(StealthPlugin())
puppeteer.use(
  RecaptchaPlugin({
    provider: {
      id: '2captcha',
      token: 'xxxx',
    },
    visualFeedback: true
  })
)

const CHROMIUM_PATH = "./node_modules/puppeteer/.local-chromium/win64-818858/chrome-win/chrome.exe"

const PRODUCT_URLS = [
  'https://www.nike.com.br/Snkrs/Produto/SB-Dunk-Low-Pro/153-169-211-270386'
]

const defaultConfig = {
  args: ['--start-maximized'],
  executablePath: CHROMIUM_PATH,
  headless: false,
  defaultViewport: {
    width: 1920,
    height: 1080,
  },
}

const initBotByUrl = async (productUrl) => {
  const browser = await puppeteer.launch(defaultConfig)
  const page = await browser.newPage()

	await page.goto(productUrl)
	await page.solveRecaptchas()
}

const initBot = () => PRODUCT_URLS.forEach(
	(productUrl) => initBotByUrl(productUrl)
)

initBot()

```

**Debug console**
```
  puppeteer-extra-plugin:base:stealth Initialized. +0ms
  puppeteer-extra plugin registered stealth +0ms
  puppeteer-extra-plugin:base:recaptcha Initialized. +0ms
  puppeteer-extra-plugin:recaptcha Initialized {
  visualFeedback: true,
  throwOnError: false,
  provider: { id: '2captcha', token: 'xxxx' }
} +0ms
  puppeteer-extra plugin registered recaptcha +81ms
  puppeteer-extra dependencies missing Set(15) {
  'stealth/evasions/chrome.app',
  'stealth/evasions/chrome.csi',
  'stealth/evasions/chrome.loadTimes',
  'stealth/evasions/chrome.runtime',
  'stealth/evasions/iframe.contentWindow',
  'stealth/evasions/media.codecs',
  'stealth/evasions/navigator.hardwareConcurrency',
  'stealth/evasions/navigator.languages',
  'stealth/evasions/navigator.permissions',
  'stealth/evasions/navigator.plugins',
  'stealth/evasions/navigator.webdriver',
  'stealth/evasions/sourceurl',
  'stealth/evasions/user-agent-override',
  'stealth/evasions/webgl.vendor',
  'stealth/evasions/window.outerdimensions'
} +2ms
  puppeteer-extra-plugin:base:stealth/evasions/chrome.app Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/chrome.app +24ms
  puppeteer-extra-plugin:base:stealth/evasions/chrome.csi Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/chrome.csi +21ms
  puppeteer-extra-plugin:base:stealth/evasions/chrome.loadTimes Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/chrome.loadTimes +23ms
  puppeteer-extra-plugin:base:stealth/evasions/chrome.runtime Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/chrome.runtime +26ms
  puppeteer-extra-plugin:base:stealth/evasions/iframe.contentWindow Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/iframe.contentWindow +30ms
  puppeteer-extra-plugin:base:stealth/evasions/media.codecs Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/media.codecs +33ms
  puppeteer-extra-plugin:base:stealth/evasions/navigator.hardwareConcurrency Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/navigator.hardwareConcurrency +36ms
  puppeteer-extra-plugin:base:stealth/evasions/navigator.languages Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/navigator.languages +38ms
  puppeteer-extra-plugin:base:stealth/evasions/navigator.permissions Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/navigator.permissions +37ms
  puppeteer-extra-plugin:base:stealth/evasions/navigator.plugins Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/navigator.plugins +44ms
  puppeteer-extra-plugin:base:stealth/evasions/navigator.webdriver Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/navigator.webdriver +63ms
  puppeteer-extra-plugin:base:stealth/evasions/sourceurl Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/sourceurl +9ms
  puppeteer-extra-plugin:base:stealth/evasions/user-agent-override Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/user-agent-override +2ms
  puppeteer-extra-plugin:base:stealth/evasions/webgl.vendor Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/webgl.vendor +2ms
  puppeteer-extra-plugin:base:stealth/evasions/window.outerdimensions Initialized. +0ms
  puppeteer-extra plugin registered stealth/evasions/window.outerdimensions +36ms
  puppeteer-extra orderPlugins:before [
  'stealth',
  'recaptcha',
  'stealth/evasions/chrome.app',
  'stealth/evasions/chrome.csi',
  'stealth/evasions/chrome.loadTimes',
  'stealth/evasions/chrome.runtime',
  'stealth/evasions/iframe.contentWindow',
  'stealth/evasions/media.codecs',
  'stealth/evasions/navigator.hardwareConcurrency',
  'stealth/evasions/navigator.languages',
  'stealth/evasions/navigator.permissions',
  'stealth/evasions/navigator.plugins',
  'stealth/evasions/navigator.webdriver',
  'stealth/evasions/sourceurl',
  'stealth/evasions/user-agent-override',
  'stealth/evasions/webgl.vendor',
  'stealth/evasions/window.outerdimensions'
] +67ms
  puppeteer-extra orderPlugins:after [
  'stealth',
  'recaptcha',
  'stealth/evasions/chrome.app',
  'stealth/evasions/chrome.csi',
  'stealth/evasions/chrome.loadTimes',
  'stealth/evasions/chrome.runtime',
  'stealth/evasions/media.codecs',
  'stealth/evasions/navigator.hardwareConcurrency',
  'stealth/evasions/navigator.languages',
  'stealth/evasions/navigator.permissions',
  'stealth/evasions/navigator.plugins',
  'stealth/evasions/navigator.webdriver',
  'stealth/evasions/sourceurl',
  'stealth/evasions/user-agent-override',
  'stealth/evasions/webgl.vendor',
  'stealth/evasions/window.outerdimensions',
  'stealth/evasions/iframe.contentWindow'
] +40ms
  puppeteer-extra-plugin:recaptcha onPageCreated about:blank +0ms
  puppeteer-extra-plugin:stealth/evasions/user-agent-override onPageCreated - Will set these user agent options {
  override: {
    userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4298.0 Safari/537.36',
    acceptLanguage: 'en-US,en',
    platform: 'Win32'
  },
  opts: { userAgent: null, locale: 'en-US,en', platform: 'Win32' }
} +0ms
  puppeteer-extra-plugin:recaptcha solveRecaptchas +0ms
  puppeteer-extra-plugin:recaptcha findRecaptchas +0ms
  puppeteer-extra-plugin:stealth/evasions/sourceurl Stripping sourceURL { method: 'Runtime.evaluate' } +0ms
  puppeteer-extra-plugin:stealth/evasions/sourceurl Stripping sourceURL { method: 'Runtime.callFunctionOn' } +70ms
  puppeteer-extra-plugin:recaptcha hasRecaptchaScriptTag true +0ms
  puppeteer-extra-plugin:recaptcha waitForRecaptchaClient - start 2021-01-23T19:40:06.770Z +0ms
  puppeteer-extra-plugin:stealth/evasions/sourceurl Stripping sourceURL { method: 'Runtime.callFunctionOn' } +11ms
  puppeteer-extra-plugin:recaptcha solveRecaptchas {
  captchas: [],
  solutions: [],
  solved: [],
  error: 'TimeoutError: waiting for function failed: timeout 10000ms exceeded'
} +0ms
```

**Versions**

```
  System:
    OS: Windows 10 10.0.18363
    CPU: (16) x64 AMD Ryzen 7 2700 Eight-Core Processor
    Memory: 9.55 GB / 15.93 GB
  Binaries:
    Node: 14.15.4 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.10 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    puppeteer: ^5.5.0 => 5.5.0 
    puppeteer-core: ^5.5.0 => 5.5.0 
    puppeteer-extra: ^3.1.16 => 3.1.16 
    puppeteer-extra-plugin-anonymize-ua: ^2.2.15 => 2.2.15 
    puppeteer-extra-plugin-recaptcha: ^3.3.1 => 3.3.1 
    puppeteer-extra-plugin-stealth: ^2.6.6 => 2.6.6 
```

**Comment by abokizh** [comment]
Same problem here. I use headless mode. Do you have any updates?

**Comment by igorthurow** [comment]
Hello @abokizh . Unfortully not, i gave up 😞

**Comment by berstend** [comment]
#496 brought major improvements (released as `puppeteer-extra-plugin-recaptcha@3.4.0`) and most likely fixes this issue, if not please ping.
